### PR TITLE
APIKIT-2518: Improve test coverage OData Extension

### PR DESCRIPTION
**Pull request by dcourtil**
Added some unit tests to increase the coverage and removed some files that are no longer used (please review this to confirm)

### CODE REVIEW

File: src/main/java/org/mule/module/apikit/odata/metadata/model/entities/EntityDefinitionProperty.java [hunk 1]
```diff
@@ -47,16 +47,16 @@ public String toString() {
...
 
     if (this.defaultValue != null) {
-      ret.append("\"defaultValue:\"" + this.defaultValue + "\",");
+      ret.append("\"defaultValue\":\"" + this.defaultValue + "\",");
     }
     if (this.maxLength != null) {
-      ret.append("\"maxLength:\"" + this.maxLength + "\",");
+      ret.append("\"maxLength\":\"" + this.maxLength + "\",");
     }
     //		if (this.fixedLength != null) {
```
Comment (JuanAller) [inline src/main/java/org/mule/module/apikit/odata/metadata/model/entities/EntityDefinitionProperty.java]: remove this comment

File: src/test/java/org/mule/module/apikit/model/EntityModelParserTestCase.java [hunk 2]
```diff
@@ -49,4 +51,21 @@ public void testPositive3() throws IOException, EntityModelParsingException {
     Assert.assertEquals("ordersId", entities.get(1).getIdElementName());
   }
 
+  @Test
+  public void testNegative() throws IOException, EntityModelParsingException {
```
Comment (JuanAller) [inline src/test/java/org/mule/module/apikit/model/EntityModelParserTestCase.java]: why testNegative and testNegative2 are differents ?

File: src/test/java/org/mule/module/apikit/odata/util/UriInfoImplTestCase.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import org.junit.Test;
+
+public class UriInfoImplTestCase {
+
+
+  @Test
+  public void getMatchUrisTest() {
+    UriInfoImpl uriInfo = new UriInfoImpl("");
+    uriInfo.getMatchedURIs();
```
Comment (JuanAller) [inline src/test/java/org/mule/module/apikit/odata/util/UriInfoImplTestCase.java]: assert

File: src/main/java/org/mule/module/apikit/odata/metadata/model/entities/EntityDefinitionProperty.java (hunk 1 above)
Comment (dcourtil) [reply src/main/java/org/mule/module/apikit/odata/metadata/model/entities/EntityDefinitionProperty.java]: Done

File: src/test/java/org/mule/module/apikit/model/EntityModelParserTestCase.java (hunk 2 above)
Comment (dcourtil) [reply src/test/java/org/mule/module/apikit/model/EntityModelParserTestCase.java]: Renamed

File: src/test/java/org/mule/module/apikit/odata/util/UriInfoImplTestCase.java (hunk 3 above)
Comment (dcourtil) [reply src/test/java/org/mule/module/apikit/odata/util/UriInfoImplTestCase.java]: My bad, unnecessary test. Removed

**Pull request merged**